src/libs/actions/Report.js

```diff
@@ -610,7 +610,7 @@ function handleReportChanged(report) {
 
     // A report can be missing a name if a comment is received via pusher event
     // and the report does not yet exist in Onyx (eg. a new DM created with the logged in person)
-    if (report.reportName === undefined) {
+    if (report.reportID && report.reportName === undefined) {
```

Comment:
Should we attempt to fetch data with ReportID but undefined reportName?

Reply:
Yeah so, it's completely OK to do that (see the comment above) and in fact we _want to_ because we might be handling a report comment from Pusher and we need to download it. Reports that we've downloaded already should have some kind of report name in storage. This is a little weird but it's just how it works for now. Maybe in the future we will fix this to make it more obvious that a report has not been fetched from the server and won't use the `reportName` implementation detail.

Reply:
How would it be possible for there to be a report without a reportID? Especially since the Onyx key uses the report ID in the name of the key?

Reply:
In theory, this could happen anytime we call `merge()` for a report that doesn't exist without also including the `reportID`. I'm not _exactly_ sure what situation leads to that happening here, but this seems like a safe way to avoid calling an API that is guaranteed to fail.